### PR TITLE
SBT integration

### PR DESCRIPTION
**Pull request by malcolm-dsider**
Slender Body Theory (SBT) alternative reservoir/wells/economics integrated into GEOPHIRES. It handless the Eavor Loop 1.0 and 2.0 styles. Test files that reproduce the results for the low-T and High-t cases from the paper:  Techno-Economic Performance of Eavor-Loop 2.0 Koenraad F. Beckers, Henry E. Johnston (2022) are included. This paper is the basis for the SBT algorithm. It was converted from MatLab code published with paper.

### CODE REVIEW

File: tests/examples/example_SBT_Lo_T.out [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+ GEOPHIRES Version: 3.5.2
+ Simulation Date: 2024-09-02
+ Simulation Time:  14:35
+ Calculation Time:     45.642 sec
+
+                           ***SUMMARY OF RESULTS***
+
+      End-Use Option: Electricity
+      Average Net Electricity Production:                     0.00 MW
+      Electricity breakeven price:                        18339.44 cents/kWh
```
Comment (softwareengineerprogrammer) [inline tests/examples/example_SBT_Lo_T.out:16]: Is this the correct/intended result for this output?

File: src/geophires_x/SBTEconomics.py [hunk 2]
```diff
@@ -0,0 +1,894 @@
...
+        cost_of_lateral_section = casing_factor * num_lateral_sections * cost_per_section
+    elif well_correlation is WellDrillingCostCorrelation.VERTICAL_LARGE_IDEAL:
+        cost_per_section = (-0.00240 * length_per_section_m ** 2 + 752.93946 * length_per_section_m + 524337.65380) * 1E-6
+        cost_of_lateral_section = casing_factor * num_lateral_sections * cost_per_section
+    elif well_correlation is WellDrillingCostCorrelation.DEVIATED_SMALL_IDEAL:
+        cost_per_section = (0.00719 * length_per_section_m ** 2 + 455.85233 * length_per_section_m + 753377.73080) * 1E-6
+        cost_of_lateral_section = casing_factor * num_lateral_sections * cost_per_section
+    elif well_correlation is WellDrillingCostCorrelation.DEVIATED_LARGE_IDEAL:
+        cost_per_section = (0.00376 * length_per_section_m ** 2 + 762.52696 * length_per_section_m + 765103.07690) * 1E-6
+        cost_of_lateral_section = casing_factor * num_lateral_sections * cost_per_section
```
Comment (softwareengineerprogrammer) [inline src/geophires_x/SBTEconomics.py:121]: Can this just be replaced with 

```python
else:
   cost_per_section = well_correlation.calculate_cost_MUSD(length_per_section_m)
   cost_of_lateral_section = casing_factor * num_lateral_sections * cost_per_section
```

File: src/geophires_x/Units.py [hunk 3]
```diff
@@ -60,6 +61,13 @@ class Units(IntEnum):
     POWERPERUNITVOLUME = auto()
     DECAY_RATE=auto()
     INFLATION_RATE=auto()
+    DYNAMIC_VISCOSITY = auto()
+
+
+class AngleUnit(str, Enum):
+    """Temperature Units"""
```
Comment (softwareengineerprogrammer) [inline src/geophires_x/Units.py:68]: Incorrect comment

File: src/geophires_x/Units.py [hunk 4]
```diff
@@ -351,3 +359,8 @@ class Decay_RateUnit(str,Enum):
 class Inflation_RateUnit(str,Enum):
     """Decay rate Units"""
     KPASCALPERYEAR = "kPa/yr"
+
+
+class Dynamic_ViscosityUnit(str,Enum):
+    """Decay rate Units"""
```
Comment (softwareengineerprogrammer) [inline src/geophires_x/Units.py:365]: Incorrect comment

File: tests/examples/example_SBT_Hi_T.out [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+                           ***SUMMARY OF RESULTS***
+
+      End-Use Option: Electricity
+      Average Net Electricity Production:                     9.62 MW
+      Electricity breakeven price:                           18.23 cents/kWh
+      Number of production wells:                             1
+      Number of injection wells:                              1
+      Flowrate per production well:                          80.0 kg/sec
+      Well depth (or total length, if not vertical):          5.8 kilometer
+      Geothermal gradient:                                     60 degC/km
```
Comment (softwareengineerprogrammer) [inline tests/examples/example_SBT_Hi_T.out:21]: Is this accurate/expected? Seems like very low output for that depth, gradient, and flow rate...

File: tests/examples/example_SBT_Lo_T.txt [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+Injection Temperature, 24
+
+***************Economics****************
+Economic Model, 3
+Reservoir Stimulation Capital Cost, 0
+Exploration Capital Cost, 0
+Starting Electricity Sale Price, 0.19
+Ending Electricity Sale Price, 0.19
+
+xSBT Generate Wireframe Graphics, True
```
Comment (softwareengineerprogrammer) [inline tests/examples/example_SBT_Lo_T.txt:50]: remove

**Review by softwareengineerprogrammer — CHANGES_REQUESTED** [review]
Great that Actions/tests are passing - a few questions and minor errors to fix

File: tests/examples/example_SBT_Lo_T.out (hunk 1 above)
Comment (malcolm-dsider) [reply tests/examples/example_SBT_Lo_T.out:16]: yes, the Lo-T version is just to validate the reservoir model, not the economics (as it says in the paper).

File: src/geophires_x/SBTEconomics.py (hunk 2 above)
Comment (malcolm-dsider) [reply src/geophires_x/SBTEconomics.py:121]: yes, good suggestion. done.

**Review by softwareengineerprogrammer — APPROVED** [review]
Looks good from code perspective now - suggest @kfbeckers review for model accuracy before merging

**Comment by softwareengineerprogrammer** [comment]
TODO post-merge: audit whether any of the following issues have been addressed by this PR:
1. https://github.com/NREL/GEOPHIRES-X/issues/278
2. https://github.com/NREL/GEOPHIRES-X/issues/279
3. https://github.com/NREL/GEOPHIRES-X/issues/280
4. https://github.com/NREL/GEOPHIRES-X/issues/281

**Review by kfbeckers — APPROVED** [review]
Great addition to GEOPHIRES!

**Pull request merged**